### DISCUSSION TITLE
RPi 4 Fresh install - ioctl() failed

### DISCUSSION THREAD

**Discussion by mwolfeu** [bluenviron/mediamtx#2370] [Questions]
I made a fresh install yesterday with a stock config on my RPi 4 (128MB GPU RAM).  WebRTC works great for > 24hrs.
I changed rpiCameraBitrate: 1000000 to be 100000.  The quality is still surprisingly good for a ~14KB stream (640x480@15FPS) .
That said, (often correlating with motion) the server dies repeatedly with "ioctl() failed" within < 5min.
Thanks in advance for any feedback.

Michael

Example run:
$ ~/mediamtx ~/mediamtx.yml 
2023/09/19 10:48:33 INF MediaMTX v1.1.0
2023/09/19 10:48:33 INF [path cam] [RPI Camera source] started
2023/09/19 10:48:33 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/09/19 10:48:33 INF [RTMP] listener opened on :1935
2023/09/19 10:48:33 INF [HLS] listener opened on :8888
2023/09/19 10:48:33 INF [WebRTC] listener opened on :8889 (HTTP)
2023/09/19 10:48:33 INF [SRT] listener opened on :8890 (UDP)
[18:41:14.531106933] [8103]  INFO Camera camera_manager.cpp:297 libcamera v0.0.5+83-bde9b04f
[18:41:14.563789853] [8104]  INFO RPI vc4.cpp:437 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media2
[18:41:14.564962502] [8103]  INFO Camera camera.cpp:1033 configuring streams: (0) 640x480-YUV420
[18:41:14.565676677] [8104]  INFO RPI vc4.cpp:565 Sensor: /base/soc/i2c0mux/i2c@1/ov5647@36 - Selected sensor format: 640x480-SGBRG10_1X10 - Selected unicam format: 640x480-pGAA
2023/09/19 10:48:34 INF [path cam] [RPI Camera source] ready: 1 track (H264)
2023/09/19 10:48:37 INF [WebRTC] [session cf8b6e85] created by 192.168.3.132:39312
2023/09/19 10:48:37 INF [WebRTC] [session cf8b6e85] peer connection established, local candidate: host/udp/192.168.3.230/48641, remote candidate: prflx/udp/192.168.3.132/35671
2023/09/19 10:48:37 INF [WebRTC] [session cf8b6e85] is reading from path 'cam', 1 track (H264)
encoder_encode(): ioctl(VIDIOC_QBUF) failed
encoder_encode(): ioctl(VIDIOC_QBUF) failed
encoder_encode(): ioctl(VIDIOC_QBUF) failed
encoder_encode(): ioctl(VIDIOC_QBUF) failed
encoder_encode(): ioctl(VIDIOC_QBUF) failed
encoder_encode(): ioctl(VIDIOC_QBUF) failed
^C2023/09/19 10:48:02 INF shutting down gracefully
2023/09/19 10:48:02 INF [SRT] listener is closing
2023/09/19 10:48:02 INF [WebRTC] listener is closing
2023/09/19 10:48:02 INF [WebRTC] [session b543472c] closed: terminated
2023/09/19 10:48:02 INF [HLS] listener is closing
2023/09/19 10:48:02 INF [RTMP] listener is closing
2023/09/19 10:48:02 INF [RTSP] listener is closing
2023/09/19 10:48:02 INF [path cam] [RPI Camera source] stopped: path is closing
2023/09/19 10:48:02 INF waiting for external commands

**Comment by aler9** [comment]
Hello, the "ioctl failed" errors happens in case of communication errors between the camera and the Raspberry. it usually happens when the cable, that is very fragile, is moved. Check that the connectors are clean and try using a case.

**Reply by mwolfeu** [reply]
It was sitting on a shelf where I didn't touch it between config modifications.

**Reply by aler9** [reply]
then make sure that GPU has at least 128MB and that RAM doesn't fill up.

**Reply by mwolfeu** [reply]
So I was able to move my cable and make it fail.   The problem I see is that restarting mediamtx will make it work fine again, but once mediamtx is caught in the ioctl death loop it can't get out on its own.   Is there a possibility to re-init the dev from scratch so I don't have to restart the whole server?

[a comment by an automated account is not shown]
